packages/fxa-js-client/client/FxAccountClient.js

```diff
@@ -118,71 +120,75 @@ FxAccountClient.VERSION = VERSION;
  * @return {Promise} A promise that will be fulfilled with JSON `xhr.responseText` of the request
  */
-FxAccountClient.prototype.signUp = function(email, password, options) {
-  var self = this;
+FxAccountClient.prototype.signUp = async function(
+  email,
+  password,
+  options = {}
+) {
+  required(email, 'email');
+  required(password, 'password');
 
-  return Promise.resolve()
-    .then(function() {
-      required(email, 'email');
-      required(password, 'password');
+  const creds = await credentials.setup(email, password);
 
-      return credentials.setup(email, password);
-    })
-    .then(function(result) {
-      /*eslint complexity: [2, 13] */
-      var endpoint = '/account/create';
-      var data = {
-        email: result.emailUTF8,
-        authPW: sjcl.codec.hex.fromBits(result.authPW),
-      };
-      var requestOpts = {};
+  let endpoint = '/account/create';
+  const data = {
+    email: creds.emailUTF8,
+    authPW: sjcl.codec.hex.fromBits(creds.authPW),
+  };
 
-      if (options) {
-        if (options.service) {
-          data.service = options.service;
-        }
+  const requestOpts = {};
 
-        if (options.redirectTo) {
-          data.redirectTo = options.redirectTo;
-        }
+  if (options.service) {
+    data.service = options.service;
+  }
 
-        // preVerified is used for unit/functional testing
-        if (options.preVerified) {
-          data.preVerified = options.preVerified;
-        }
+  if (options.redirectTo) {
+    data.redirectTo = options.redirectTo;
+  }
 
-        if (options.resume) {
-          data.resume = options.resume;
-        }
+  // preVerified is used for unit/functional testing
+  if (options.preVerified) {
+    data.preVerified = options.preVerified;
+  }
 
-        if (options.keys) {
-          endpoint += '?keys=true';
-        }
+  if (options.resume) {
+    data.resume = options.resume;
+  }
 
-        if (options.lang) {
-          requestOpts.headers = {
-            'Accept-Language': options.lang,
-          };
-        }
+  if (options.keys) {
+    endpoint += '?keys=true';
+  }
 
-        if (options.metricsContext) {
-          data.metricsContext = metricsContext.marshall(options.metricsContext);
-        }
+  if (options.lang) {
+    requestOpts.headers = {
+      'Accept-Language': options.lang,
+    };
+  }
 
-        if (options.style) {
-          data.style = options.style;
-        }
-      }
+  if (options.metricsContext) {
+    data.metricsContext = metricsContext.marshall(options.metricsContext);
+  }
 
-      return self.request
-        .send(endpoint, 'POST', null, data, requestOpts)
-        .then(function(accountData) {
-          if (options && options.keys) {
-            accountData.unwrapBKey = sjcl.codec.hex.fromBits(result.unwrapBKey);
-          }
-          return accountData;
-        });
-    });
+  if (options.style) {
+    data.style = options.style;
+  }
+
+  if (options.verificationMethod) {
```

Comment:
This is the only changed part in the `.signUp` function, everything else is just syntax updates.